core/Constants.php

```diff
@@ -104,7 +123,12 @@ function stripslashes_recursively(&$array) {
 			if(SS_TRUSTED_PROXY_IPS === '*') {
 				$trusted = true;
 			} elseif(isset($_SERVER['REMOTE_ADDR'])) {
```

Comment:
I think you should refactor this to an `ip_in_any_range()` function, then you can add tests of that function.

```php
if(SS_TRUSTED_PROXY_IPS === '*') {
  $trusted = true;
} elseif(isset($_SERVER['REMOTE_ADDR'])) {
  $trusted = ip_in_any_range($_SERVER['REMOTE_ADDR'], SS_TRUSTED_PROXY_IPS);
}
```

Ensure there is a test for exact match on `::1`. [Handling of IPv6 subnets](https://www.ibm.com/support/knowledgecenter/STCMML8/com.ibm.storage.ts3500.doc/opg_3584_IPv4_IPv6_prefix_subnet_mask.html) wouldn't block merge but it might be nice.